core/src/main/scala/org/apache/spark/util/AkkaUtils.scala

```diff
@@ -228,9 +230,22 @@ private[spark] object AkkaUtils extends Logging {
       actorSystem: ActorSystem): ActorRef = {
     val executorActorSystemName = SparkEnv.executorActorSystemName
     Utils.checkHost(host, "Expected hostname")
-    val url = s"akka.tcp://$executorActorSystemName@$host:$port/user/$name"
+    val url = address(executorActorSystemName, host, port, name, conf)
     val timeout = AkkaUtils.lookupTimeout(conf)
     logInfo(s"Connecting to $name: $url")
     Await.result(actorSystem.actorSelection(url).resolveOne(timeout), timeout)
   }
+
+  def protocol(conf: SparkConf): String = {
+    if (conf.getBoolean("spark.ssl.enabled", defaultValue = false)) {
+      "akka.ssl.tcp"
+    } else {
+      "akka.tcp"
+    }
+  }
+
+  def address(systemName: String, host: String, port: Any, actorName: String,
```

Comment:
Minor style issue: if the arguments won't fit on one line, then wrap them one per line, e.g.

```
def address(
    systemName: String,
    host: String,
    [...]
```